src/courses/tests/test_models.py

```diff
@@ -340,6 +340,13 @@ def test_xp_per_day_ave(self, days_so_far):
         xp_per_day = self.course_student.xp_per_day_ave()
         self.assertEqual(xp_per_day, 0)
 
+    @patch('profile_manager.models.Profile.xp_per_course')
+    def test_student_with_negative_xp(self, xp_per_course):
+        """Test if an assertion error is raised when there is a student with negative xp"""
+        xp_per_course.return_value = -10
+        self.assertRaises(AssertionError, CourseStudent.objects.calc_semester_grades,
+                          Semester.objects.get_current())
+
```

Comment:
Excellent!  Can you also test that the View that calls this method returns successfully in cases where there is negative XP, and that the warning is displayed on the page?  Currently it returns 403 or 404.